src/devices/Pn532/Pn532.cs

```diff
@@ -0,0 +1,1623 @@
+    /// PN532 RFID/NFC reader
+    /// </summary>
+    public class Pn532 : RfidWriteRead, IDisposable
+    {
+        // Timeout 
+        private const int ReadTimeOut = 500;
+        private const int SerialReadTimeOut = 500;
+        // Communication way
+        private const byte ToHostCheckSumD5 = 0xD5;
+        private const byte FromHostCheckSumD4 = 0xD4;
+        // Preamble, codes and postamble
+        private const byte Preamble = 0x00;
+        private const byte Postamble = 0x00;
+        private const byte StartCode1 = 0x00;
+        private const byte StartCode2 = 0xFF;
+        // Operation type for SPI
+        private const byte WriteData = 0b0000_0001;
+        private const byte ReadStatus = 0b0000_0010;
+        private const byte ReadData = 0b0000_0011;
+        // Acknowledge
+        private byte[] AckBuffer = { 0x00, 0x00, 0xFF, 0x00, 0xFF, 0x00 };
+        // Specific buffer to wake up the sensor in serial HSU mode
+        private byte[] SerialWakeUp = new byte[] { 0x55, 0x55, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+        private byte[] I2cWakeUp = new byte[] { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+        private int I2cMAxBuffer = 1024;
+        private ParametersFlags _parametersFlags;
+
+        private SpiDevice _spiDevice = null;
+        private I2cDevice _i2cDevice = null;
+        private GpioController _controller = null;
+        private SerialPort _serialPort = null;
+        private int _pin = 18;
+        private SecurityAccessModuleMode _securityAccessModuleMode;
+        private uint _virtualCardTimeout = 20;
+
+        /// <summary>
+        /// The Log level
+        /// </summary>        
+        public LogLevel LogLevel { get { return LogInfo.LogLevel; } set { LogInfo.LogLevel = value; } }
+
+        /// <summary>
+        /// The location to log the info
+        /// </summary>
+        public LogTo LogTo { get { return LogInfo.LogTo; } set { LogInfo.LogTo = value; } }
+
+        /// <summary>
+        /// Firmware version information
+        /// </summary>
+        public FirmwareVersion FirmwareVersion { get; internal set; }
+
+        #region Spi and I2c Settings
+        /// <summary>
+        /// PN532 SPI Clock Frequency
+        /// </summary>
+        public const int SpiClockFrequency = 2_000_000;
+
+        /// <summary>
+        /// Only SPI Mode supported is Mode0
+        /// </summary>
+        public const SpiMode SpiMode = System.Device.Spi.SpiMode.Mode0;
+
+        /// <summary>
+        /// The default I2C address
+        /// </summary>
+        public const byte I2cDefaultAddress = 0x24;
+
+        #endregion        
+
+        /// <summary>
+        /// Create a PN532 using Serial Port
+        /// </summary>
+        /// <param name="portName">The port name</param>
+        public Pn532(string portName)
+        {
+#if DEBUG
```

Comment:
I'd rather see this set in a test directly rather than ifdef this.

Reply:
I was planning to remove this further. I will add it to the constructor which will allow to set it right at the constructor level. You may not want a debug view if you are in debug mode.